### PR TITLE
Cleanup _merge_embed

### PR DESCRIPTION
**Pull request by dzbarsky**
**What type of PR is this?**
Starlark cleanup

**What does this PR do? Why is it needed?**
Since only one the lib or one of the embeds can specify cgo, we can simplify this a bit

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

File: go/private/context.bzl [hunk 1]
```diff
@@ -259,9 +261,7 @@ def _library_to_source(go, attr, library, coverage_instrumented):
     srcs = attr_srcs + generated_srcs
     embedsrcs = [f for t in getattr(attr, "embedsrcs", []) for f in as_iterable(t.files)]
     attr_deps = getattr(attr, "deps", [])
-    generated_deps = getattr(library, "deps", [])
-    deps = attr_deps + generated_deps
-    deps = [get_archive(dep) for dep in deps]
+    deps = [get_archive(dep) for dep in attr_deps]
```
Comment (fmeum) [inline go/private/context.bzl:264]: Why is this change safe?

Comment (dzbarsky) [reply go/private/context.bzl:264]: currently the only supported way to make a `GoLibrary` is with `new_library`, which does not have a `deps` attribute. While it does take a `kwargs` and assign anything from it into the provider, `deps` is not documented in either `GoLibrary` (providers.rst) or `new_library` (toolchains.rst), so I consider this to be relying on undocumented behavior.

In terms of impact, in order to be affected by this change, code would need to:
1. Be passing `ctx.attr` to `attr` in `_library_from_source` but having additonal deps that are not in `ctx.attr.deps`
2. Be passing a custom struct to `attr` in `_library_from_source` and wanting to have deps.

The 2nd case is trivially fixable by passing deps, and the first case is also fixable by switching to constructing a struct. It seems like a fairly contrived scenario though.

We could make the workaround even simpler by taking an explicit `deps` in `_library_to_source` and falling back to `getattr` if it's not passed?

Comment (dzbarsky) [reply go/private/context.bzl:264]: I'm not sure why this ended up in this PR, I can split to a separate one if you prefer

**Comment by fmeum** [comment]
This has conflicts

**Comment by dzbarsky** [comment]
> This has conflicts

rebased!

**Review by fmeum — APPROVED** [review]

**Pull request merged**